plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -51,19 +54,30 @@ class BigQueryConnectionMethod(StrEnum):
     OAUTH = 'oauth'
     SERVICE_ACCOUNT = 'service-account'
     SERVICE_ACCOUNT_JSON = 'service-account-json'
+    OAUTH_SECRETS = 'oauth-secrets'
```

Comment:
Were you going to rename this to `refresh-token`?

Reply:
i decided not to! I actually updated this PR to also accept an access token directly via the `token` configuration. So, it's kind of just a grab bag of oauth auth methods. Google lets you supply:
 - a client id / client secret / refresh token / token uri OR
 - a standalone access token

The access token is going to expire after ~60 mins (not well defined), but it was a one-line change and feels like a reasonable enough use case to support. You buy it?

Reply:
gotcha — does that mean that this resolves #2802 as well?